### PR TITLE
[DEV-6466] Update Total Budgetary Resources Calculation

### PR DESCRIPTION
**Pull request by collinwr**
**Description:**
This PR updates the logic to calculate the `total_budget_authority`, or Total Budgetary Resources, on the disaster overview endpoint. 

**NOTE**
This PR has a dependency on https://github.com/fedspendingtransparency/usaspending-api/pull/2867 Do not merge this PR until that PR has been merged. 

**Technical details:**
This PR utilizes the new field `budget_authority_unobligated_balance_brought_forward_cpe` on the `gtas_sf133_balances` table to correct the Total Budget Authority calculation. This part of the API response previously only included an aggregation of the `total_budgetary_resources_cpe` fields of the GTAS entries for the latest submission. It now subtracts the aggregation of the aforementioned `budget_authority_unobligated_balance_brought_forward_cpe` fields from this value. 

**Requirements for PR merge:**

1. [ ] Unit & integration tests updated - Unit tests are passing, but are still being updated. 
2. N/A API documentation updated
3. [ ] Necessary PR reviewers:
    - [x] Backend
    - [ ] Frontend <OPTIONAL>
    - [ ] Operations <OPTIONAL>
    - [ ] Domain Expert <OPTIONAL>
4. [x] Matview impact assessment completed
5. [x] Frontend impact assessment completed
6. [ ] Data validation completed
7. N/A Appropriate Operations ticket(s) created
8. [ ] Jira Ticket [DEV-6566](https://federal-spending-transparency.atlassian.net/browse/DEV-6466):
    - [x] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/disaster/v2/views/overview.py [hunk 1]
```diff
@@ -57,6 +58,18 @@ def funding(self):
             .values("def_code", "amount")
         )
 
+    def unobligated_balance(self):
```
Comment (tony-sappe) [inline usaspending_api/disaster/v2/views/overview.py:61]: Is there a reason `budget_authority_unobligated_balance_brought_forward_cpe` couldn't be included in the original funding ORM query? Seems like the annotation could include a sum for both fields

Comment (collinwr) [reply usaspending_api/disaster/v2/views/overview.py]: I have combined these queries into one for clarity and performance.

File: usaspending_api/disaster/v2/views/overview.py [hunk 2]
```diff
@@ -105,3 +115,6 @@ def total_outlays(self):
             .aggregate(total=Sum("gross_outlay_amount_by_tas_cpe"))["total"]
             or 0.0
         )
+
+    def sum_values(self, obj, key):
```
Comment (tony-sappe) [inline usaspending_api/disaster/v2/views/overview.py:119]: note for future work

Technically `obj` is a list of dictionaries, not an object. For a tiny function like this, I would probably make it a staticmethod, use slightly more descriptive argument names, and add type hints for additional communication. None of this is blocking the PR from being merged

**Review by tony-sappe — APPROVED** [review]

File: usaspending_api/disaster/v2/views/overview.py (hunk 2 above)
Comment (collinwr) [reply usaspending_api/disaster/v2/views/overview.py]: I went ahead and did this

**Review by tony-sappe — APPROVED** [review]

**Pull request merged**